src/apis/stockx/asks/index.js

```diff
@@ -0,0 +1,140 @@
+import moment from 'moment';
+
+import Api from '../../base';
+import { randomInclusive } from '../../../utils';
+
+export default class AsksApi extends Api {
+  // TODO!
+  async list(options = {}) {}
+
+  async update(ask = {}, options = {}) {
+    const { amount } = options;
+    const { headers, bearer, jar, proxy, currency } = this.data;
+    const expiresAt = moment().add(30, 'days').utc().format();
+
+    let chainId;
+    let skuUuid;
+    try {
+      ({ chainId, skuUuid } = ask);
+
+      if (!amount || !skuUuid || !chainId) {
+        const error = new Error('Invalid amount product id, and/or ask id!');
+        error.status = 404;
+        throw error;
+      }
+
+      if (!bearer) {
+        const error = new Error('Please login first!');
+        error.status = 401;
+        throw error;
+      }
+
+      const res = await this._request('https://stockx.com/api/portfolio?a=ask', {
+        headers: {
+          ...headers,
+          authorization: `Bearer ${bearer}`,
+          'content-type': 'application/json',
+        },
+        jar,
+        json: {
+          PortfolioItem: {
+            localAmount: amount,
+            skuUuid,
+            localCurrency: currency,
+            expiresAt: expiresAt,
+            chainId,
+          },
+        },
+        method: 'POST',
+        proxy,
+        resolveWithFullResponse: true,
+        simple: false,
+      });
+
+      const { statusCode, body } = res;
+      if (!statusCode || (statusCode && statusCode !== 200)) {
+        const err = new Error('Invalid response code!');
+        if (statusCode === 400) {
+          err.message = 'Account on hold!';
```

Comment:
Is this the best message? Maybe `Account suspended`?

Reply:
@matthew1232 lemme know your thoughts

Reply:
I am not completely familiar with their api, is `Bad Request` indicative of an existing account that no longer works?

Reply:
If the stringified JSON response includes 'account suspended' or whatever the body is, send the error "Account suspended". Same with includes "invalid email" throwing "Invalid email/password", which I had in my previous version.

Reply:
The 400 response will happen if the email/password is incorrect as well.

Reply:
Ah okay gotcha. I think I’ll amend that back in then, otherwise I’ll just send a generic “Bad Request” message back.

Reply:
If that's the case, then when you refactor the "other error handling" in a different PR, you could make a strategy factory that deals with all the 3rd party api status codes and creates `Error`'s for your API.

Reply:
Resolved. Added a central error throwing mechanism that includes just enough details to the user while keeping the stack under wrap.